src/runtime_src/core/edge/drm/zocl/zocl_bo.c

```diff
@@ -141,12 +142,50 @@ zocl_create_bo(struct drm_device *dev, uint64_t unaligned_size, u32 user_flags)
 		err = drm_gem_object_init(dev, &bo->gem_base, size);
 		if (err < 0)
 			goto free;
-	} else {
+	} else if (user_flags & XCL_BO_FLAGS_CMA) {
+		/* Allocate from CMA buffer */
 		cma_obj = drm_gem_cma_create(dev, size);
 		if (IS_ERR(cma_obj))
 			return ERR_PTR(-ENOMEM);
 
 		bo = to_zocl_bo(&cma_obj->base);
+	} else {
+		/* We are allocating from a separate BANK, i.e. PL-DDR */
+		unsigned int bank = GET_MEM_BANK(user_flags);
+
```

Comment:
Check for valid bank index?

Reply:
Actually, we won't be here if the bank is invalid. The bank validation was done in zocl_create_bo_ioctl(). But to be safe, I followed your suggestion adding a check anyway in case zocl_create_bo () will be called from other pathes than zocl_create_bo_ioctl() in the future.